onadata/apps/main/views.py

```diff
@@ -1423,6 +1424,41 @@ def enketo_preview(request, username, id_string):
     return HttpResponseRedirect(enketo_preview_url)
 
 
+def service_health(request):
+    """
+    This endpoint checks whether the various services(Database, Cache, e.t.c )
+    of the application are running as expected. Returns a 200 Status code if
+    all is well and a 500 if a service is down
+    """
+    service_degraded = False
+    service_statuses = {}
+
+    # Check if Database connections are present & data is retrievable
+    for database in getattr(settings, 'DATABASES').keys():
+        try:
+            XForm.objects.using(database).first()
+        except Exception as e:
+            service_statuses[f'{database}-Database'] = f'Degraded state; {e}'
+            service_degraded = True
+        else:
+            service_statuses[f'{database}-Database'] = 'OK'
+
+    # Check if cache is accessible
+    try:
+        cache.set('ping', 'pong')
+        service_degraded = not (cache.get('ping') == 'pong')
```

Comment:
Do we really want to restart a pod or create a new pod when we don't have access to the cache?

Reply:
I believe so; most of the processing intensive functionalities like user permission generation & other project calculations are stored and tracked using the cache. I think if their down we'd experience some impact on the service... though this is just an assumption

Reply:
It is better to have a degraded service than not to have one all. Because if we lose the cache we will lose the whole application, kubernetes will mark all the pod unhealthy